shell/platform/darwin/macos/framework/Source/FlutterCompositor.mm

```diff
@@ -98,24 +101,32 @@
   [view.surfaceManager presentSurfaces:surfaces
                                 atTime:presentation_time
                                 notify:^{
-                                  PresentPlatformViews(view, *platform_views_layers);
+                                  // Gets a presenter or create a new one for the view.
+                                  ViewPresenter& presenter = presenters_[view_id];
+                                  presenter.PresentPlatformViews(view, *platform_views_layers,
+                                                                 platform_view_controller_);
                                 }];
 
   return true;
 }
 
-void FlutterCompositor::PresentPlatformViews(
+FlutterCompositor::ViewPresenter::ViewPresenter()
+    : mutator_views_([NSMapTable strongToStrongObjectsMapTable]) {}
+
+void FlutterCompositor::ViewPresenter::PresentPlatformViews(
     FlutterView* default_base_view,
-    const std::vector<PlatformViewLayerWithIndex>& platform_views) {
+    const std::vector<PlatformViewLayerWithIndex>& platform_views,
+    const FlutterPlatformViewController* platform_view_controller) {
   FML_DCHECK([[NSThread currentThread] isMainThread])
       << "Must be on the main thread to present platform views";
 
   // Active mutator views for this frame.
   NSMutableArray<FlutterMutatorView*>* present_mutators = [NSMutableArray array];
 
   for (const auto& platform_view : platform_views) {
-    [present_mutators addObject:PresentPlatformView(default_base_view, platform_view.first,
```

Comment:
Just a nit - The formatting looks bit weird after adding additional argument. Maybe use a temporary variable for the result of PresentPlatformView?